Sprintfinity3902/Game1.cs

```diff
@@ -82,129 +97,120 @@ protected void Reset()
-
-
-            base.Update(gameTime);
         }
 
         protected override void Draw(GameTime gameTime)
         {
+            base.Draw(gameTime);
+
             GraphicsDevice.Clear(Color.Black);
             SpriteBatch.Begin(SpriteSortMode.Deferred, null, SamplerState.PointClamp);
 
-            dungeon.Draw(SpriteBatch);
-            pauseMenu.Draw(SpriteBatch);
-
-            foreach (IHud hud in huds)
-            {
-                foreach (IEntity icon in hud.Icons)
-                {
-                    icon.Draw(SpriteBatch, Color.White);
-                }
+            switch (State) {
+                case GameState.PAUSED:
+                case GameState.PAUSED_TRANSITION:
+                case GameState.LOSE:
+                case GameState.WIN:
+                case GameState.PLAYING:
+                    
+                    foreach (IHud hud in huds) {
+                        foreach (IEntity icon in hud.Icons) {
+                            icon.Draw(SpriteBatch, Color.White);
+                        }
+                    }
+
+                    dungeon.Draw(SpriteBatch);
+
+                    playerCharacter.Draw(SpriteBatch, Color.White);
+
+                    break;
+                case GameState.OPTIONS:
+                    optionMenu.Draw(SpriteBatch);
+                    break;
             }
-
-            playerCharacter.Draw(SpriteBatch, Color.White);
-
-            boomerangItem.Draw(SpriteBatch, Color.White);
-            bombItem.Draw(SpriteBatch, Color.White);
-            movingSword.Draw(SpriteBatch, Color.White);
-            bowArrow.Draw(SpriteBatch, Color.White);
-
+            
             SpriteBatch.End();
+        }
 
-            base.Draw(gameTime);
-
+        public void UpdateState(GameState state) {
+            switch (state) {
+                case GameState.PAUSED:
+                    break;
+                case GameState.PAUSED_TRANSITION:
+                    if (State.Equals(GameState.PLAYING)) {
+                        KeyboardManager.Instance.PushCommandMatrix();
+                        KeyboardManager.Instance.RegisterKeyUpCallback(Pause, Keys.P);
+                    }
+                    break;
+                case GameState.WIN:
+                    dungeon.UpdateState(IDungeon.GameState.WIN);
+                    break;
+                case GameState.OPTIONS:
+                    optionMenu.Start();
+                    break;
+                case GameState.PLAYING:
+                    if (State.Equals(GameState.PAUSED_TRANSITION)) {
+                        KeyboardManager.Instance.PopCommandMatrix();
+                    }
+                    break;
+                case GameState.LOSE:
+                    dungeon.UpdateState(IDungeon.GameState.LOSE);
+                    break;
+            }
```

Comment:
Not a problem for this pr but this looks like a candidate for a state pattern. OOP could likely abstract away the whole switchcase. Thoughts?

Reply:
Yeah I agree here. There could be separate classes to handle each case, and we could have a CurrentState object in order to set what the CurrentState is similar to the Goriya, Player, etc. Perhaps a refactoring job if we finish Sprint 4 early.

Reply:
^ & ^^ haha